kernel/network/ethernet.c

```diff
@@ -0,0 +1,78 @@
+
+/******************************************************************************
+ *       ninjastorms - shuriken operating system                              *
+ *                                                                            *
+ *    Copyright (C) 2013 - 2016  Andreas Grapentin et al.                     *
+ *                                                                            *
+ *    This program is free software: you can redistribute it and/or modify    *
+ *    it under the terms of the GNU General Public License as published by    *
+ *    the Free Software Foundation, either version 3 of the License, or       *
+ *    (at your option) any later version.                                     *
+ *                                                                            *
+ *    This program is distributed in the hope that it will be useful,         *
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+ *    GNU General Public License for more details.                            *
+ *                                                                            *
+ *    You should have received a copy of the GNU General Public License       *
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.   *
+ ******************************************************************************/
+
+#include "ethernet.h"
+#include "kernel/network/arp.h"
+#include "kernel/network/e1000.h"
+#include "kernel/network/network_io.h"
+#include "kernel/logger/logger.h"
+
+#include <sys/types.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define INTTOHEXCHAR(val) ((val) > 9 ? (val)+'a'-10 : (val) + '0')
+
+void
+send_ethernet(mac_address_t dest_mac, ether_type eth_type, void *payload, size_t len)
+{
+  // make sure ethernet frame is at least 60 bytes long
+  if (len < 46)
+    len = 46;
+  ethernet_frame_t *eth_frame = (ethernet_frame_t *) malloc(sizeof(ethernet_frame_t) + len);
+  eth_frame->dest_mac = hton_mac(dest_mac);
+  eth_frame->src_mac = my_mac(); // mac address is stored in big endian on nc, so no swap
```

Comment:
Maybe we should change this for consistency?

Reply:
Yes, especially if one wants to e.g. print `my_mac()` it will always be in wrong order